samples/unit-tests/coloraxis/extremes/demo.js

```diff
@@ -23,3 +23,56 @@ QUnit.test('Extremes are calculated from all series. #6209', function (assert) {
     assert.strictEqual(chart.colorAxis[0].max, 600, 'Max is 600');
     assert.strictEqual(chart.colorAxis[0].min, 0, 'Min is 0');
 });
+
+QUnit.test('Color not lost after extreme change, #17945', function (assert) {
+    const chart = Highcharts.stockChart('container', {
+        chart: {
+            type: 'line'
+        },
+        colorAxis: {
+            stops: [
+                [0.25, 'red'],
+                [0.50, 'green'],
+                [1, 'blue']
+            ]
+        },
+        series: [{
+            dataGrouping: {
+                enabled: true,
+                groupPixelWidth: 15,
+                units: [
+                    [
+                        'week',
+                        [1]
+                    ]
+                ]
```

Comment:
Hint: If you want to force dataGrouping with small datasets then simply set `forced = true`. You can resign from options like units, groupPixelWidth and pointStart/Interval